**MODELLING STATISTICS**

In [ ]:

library(tidyverse) 
list.files(path = "../input")



── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──

✔ ggplot2 3.3.2     ✔ purrr   0.3.4
✔ tibble  3.0.4     ✔ dplyr   1.0.2
✔ tidyr   1.1.2     ✔ stringr 1.4.0
✔ readr   1.4.0     ✔ forcats 0.5.0

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()



[1] "world-happiness"

In [ ]:
happy <- read.csv(file = 'testdata.csv', header = TRUE) 

In [ ]:
df <- as.data.frame(happy[,c('Social', 'Healthy.life.expectancy', 'Freedom')])

In [ ]:
happy <- as.data.frame(happy[,c( 'Score', 'Social', 'Healthy.life.expectancy', 'Freedom')])

# Multiple Linear Regression

In [ ]:
fit <- lm(Score ~  Social+ Health + Freedom, data=happy)
summ_fit <- summary(fit)
summary(fit)


Call:
lm(formula = Score ~ GDP.per.capita + Social.support + Healthy.life.expectancy + 
    Freedom.to.make.life.choices + Generosity + Perceptions.of.corruption, 
    data = happy)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.75304 -0.35306  0.05703  0.36695  1.19059 

Coefficients:
                             Estimate Std. Error t value Pr(>|t|)    
(Intercept)                    1.7952     0.2111   8.505 1.77e-14 ***
GDP.per.capita                 0.7754     0.2182   3.553 0.000510 ***
Social.support                 1.1242     0.2369   4.745 4.83e-06 ***
Healthy.life.expectancy        1.0781     0.3345   3.223 0.001560 ** 
Freedom.to.make.life.choices   1.4548     0.3753   3.876 0.000159 ***
Generosity                     0.4898     0.4977   0.984 0.326709    
Perceptions.of.corruption      0.9723     0.5424   1.793 0.075053 .  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5335 on 149 degrees of freedom
Multiple R-s

**Since the p-value of Freedom is less than 0.05, it is not significant. Also, $R^2$, coefficient of determination, is a measure of goodness of fitness of the model, between 0 and 1. In this case, $R^2$ equals 0.779, which is relatively a good fit.**

**This university students data follows a Normal Distribution**

**We standarlized the coefficient and the variable Freedom has a relatively high coefficent.**

# Logistic Regression

In [ ]:
happy$Score.index <- factor(happy$Score> mean(happy$Score), levels=c(TRUE,FALSE),labels=c("High", "Low"))

Next, we generate a logistic regression model.

In [ ]:
fit.full<- glm(Score.index ~ GDP.per.capita + Social.support + Healthy.life.expectancy + Freedom.to.make.life.choices + Generosity + Perceptions.of.corruption, data=happy, family=binomial())
summary(fit.full)


Call:
glm(formula = Score.index ~ GDP.per.capita + Social.support + 
    Healthy.life.expectancy + Freedom.to.make.life.choices + 
    Generosity + Perceptions.of.corruption, family = binomial(), 
    data = happy)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-3.3001  -0.3253   0.0017   0.2008   2.1362  

Coefficients:
                             Estimate Std. Error z value Pr(>|z|)    
(Intercept)                    18.557      3.634   5.106 3.28e-07 ***
GDP.per.capita                 -1.311      1.644  -0.798  0.42508    
Social.support                 -5.812      2.022  -2.874  0.00405 ** 
Healthy.life.expectancy        -9.066      3.058  -2.964  0.00303 ** 
Freedom.to.make.life.choices   -8.106      2.603  -3.114  0.00185 ** 
Generosity                      2.915      3.471   0.840  0.40112    
Perceptions.of.corruption      -3.393      5.627  -0.603  0.54647    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for b

In this case, only Social Support, Healthy Life Expectancy and Freedom to Make Life Choices are significant, since the p-values are less than 0.05. Residual deviance measure the badness of fit of the 5 predictors. Then we need to test if the drop of deviance between the null model and our model is significant.

In [ ]:
fit.null <- glm(Score.index ~ 1, data=happy, family=binomial()) 
anova(fit.null, fit.full, test="Chisq")

Resid. Df Resid. Dev Df Deviance Pr(>Chi)    
1 155       216.2363   NA       NA           NA
2 149        78.5115    6 137.7248 3.026845e-27

In [ ]:
fit.reduced <- glm(Score.index ~ Social.support + Healthy.life.expectancy + Freedom.to.make.life.choices,
                   data = happy, family=binomial())
summary(fit.reduced)


Call:
glm(formula = Score.index ~ Social.support + Healthy.life.expectancy + 
    Freedom.to.make.life.choices, family = binomial(), data = happy)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-3.3203  -0.3579   0.0017   0.1955   2.2563  

Coefficients:
                             Estimate Std. Error z value Pr(>|z|)    
(Intercept)                    19.107      3.527   5.418 6.03e-08 ***
Social.support                 -6.262      1.873  -3.343 0.000828 ***
Healthy.life.expectancy       -10.691      2.737  -3.905 9.41e-05 ***
Freedom.to.make.life.choices   -7.448      2.292  -3.250 0.001153 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 216.24  on 155  degrees of freedom
Residual deviance:  80.78  on 152  degrees of freedom
AIC: 88.78

Number of Fisher Scoring iterations: 7


P-values shows that only Social Support, Life Expectancy and Freedom are significant, so here is a model with only these predictors. All predictors are significant in the reduced model. The AIC in this model is 88.78 which is smaller and better than the full model.


# Decision Trees

In [ ]:
library(ISLR)
set.seed(1234)
train <- sample(nrow(happy), 0.8*nrow(happy)) 
training <- happy[train,]
validation <- happy[-train,] 

In [ ]:
dtree.pred <- predict(dtree.pruned, validation, type="class")
dtree.perf <- table(validation$Score.index, dtree.pred,dnn=c("Actual", "Predicted"))
dtree.perf

      Predicted
Actual High Low
  High   18   1
  Low     0  13

# Random Forest

In [ ]:
library(ISLR)
set.seed(1234)
train <- sample(nrow(happy), 0.8*nrow(happy)) 
training <- happy[train,]

In [ ]:
library(randomForest)
set.seed(1234)
fit.forest <- randomForest(Score~  Social + Health + Freedom,data=testdata,na.action=na.roughfix, importance=TRUE)
fit.forest

randomForest 4.6-14

Type rfNews() to see new features/changes/bug fixes.


Attaching package: ‘randomForest’


The following object is masked from ‘package:dplyr’:

    combine


The following object is masked from ‘package:ggplot2’:

    margin





Call:
 randomForest(formula = Score ~ GDP.per.capita + Social.support +      Healthy.life.expectancy + Freedom.to.make.life.choices +      Generosity + Perceptions.of.corruption, data = training,      importance = TRUE, na.action = na.roughfix) 
               Type of random forest: regression
                     Number of trees: 500
No. of variables tried at each split: 2

          Mean of squared residuals: 0.2413737
                    % Var explained: 80.28